internal/controllers/machinedeployment/machinedeployment_status_test.go

```diff
@@ -0,0 +1,1185 @@
+	ownerRemediatedV1Beta2 := metav1.Condition{Type: clusterv1.MachineOwnerRemediatedV1Beta2Condition, Status: metav1.ConditionFalse, Message: "Remediation in progress"}
+
+	tests := []struct {
+		name                 string
+		machineDeployment    *clusterv1.MachineDeployment
+		machines             []*clusterv1.Machine
+		getMachinesSucceeded bool
+		expectCondition      metav1.Condition
+	}{
+		{
+			name:                 "get machines failed",
+			machineDeployment:    &clusterv1.MachineDeployment{},
+			machines:             nil,
+			getMachinesSucceeded: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineDeploymentRemediatingInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:              "Without unhealthy machines",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1"),
+				fakeMachine("m2"),
+			},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+				Status: metav1.ConditionFalse,
+				Reason: clusterv1.MachineDeploymentNotRemediatingV1Beta2Reason,
+			},
+		},
+		{
+			name:              "With machines to be remediated by KCP",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1", withConditions(healthCheckSucceeded)),    // Healthy machine
+				fakeMachine("m2", withConditions(healthCheckNotSucceeded)), // Unhealthy machine, not yet marked for remediation
+				fakeMachine("m3", withConditions(healthCheckNotSucceeded, ownerRemediated), withV1Beta2Condition(ownerRemediatedV1Beta2)),
+			},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.MachineDeploymentRemediatingV1Beta2Reason,
+				Message: "Remediation in progress from Machine m3",
+			},
+		},
+		{
+			name:              "With one unhealthy machine not to be remediated by KCP",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1", withConditions(healthCheckSucceeded)),    // Healthy machine
+				fakeMachine("m2", withConditions(healthCheckNotSucceeded)), // Unhealthy machine, not yet marked for remediation
+				fakeMachine("m3", withConditions(healthCheckSucceeded)),    // Healthy machine
+			},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  clusterv1.MachineDeploymentNotRemediatingV1Beta2Reason,
+				Message: "Machine m2 is not healthy (not to be remediated by KCP)",
```

Comment:
This message should also not contain KCP (copy & paste error in aggregateUnhealthyMachines)